src/icc-tags.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {iccTags, iccProfile} from './icc-tag-names';
+
+export default {
+    read
+};
+
+const PROFILE_HEADER_LENGTH = 84;
+
+function read(dataView, iccData) {
+    try {
+        const totalIccProfileLength = iccData.reduce((sum, icc) => sum + icc.length, 0);
+
+        const iccBinaryData = new Uint8Array(totalIccProfileLength);
+        let offset = 0;
+        const buffer = dataView.buffer;
+        for (let chunkNumber = 1; chunkNumber <= iccData.length; chunkNumber++) {
+            const iccDataChunk = iccData.find((x) => x.chunkNumber === chunkNumber);
+            if (!iccDataChunk) {
+                throw new Error(`ICC chunk ${chunkNumber} not found`);
+            }
+
+            const data = buffer.slice(iccDataChunk.offset, iccDataChunk.offset + iccDataChunk.length);
+            const chunkData = new Uint8Array(data);
+
+            iccBinaryData.set(chunkData, offset);
+            offset += chunkData.length;
+        }
+
+        return parseTags(new DataView(iccBinaryData.buffer));
+    } catch (error) {
+        return {};
+    }
+}
+
+const getContentAtOffsetAsString = (offset, buffer) => {
+    return String.fromCharCode.apply(null, new Uint8Array(buffer.slice(offset, offset + 4))).trim();
+};
+
+function sliceToString(slice) {
+    return String.fromCharCode.apply(null, new Uint8Array(slice));
+}
+
+function parseTags(dataView) {
```

Comment:
This function has a lot of magic numbers. Could they get their own constants with some descriptive names?

Reply:
should be more readable now